src/Clay/Grid.hs

```diff
@@ -1,58 +1,249 @@
+import Control.Monad (when)
+
 
 -- | Property sets the gaps (gutters) between rows and columns.
-gridGap :: Size a -> Css
-gridGap = key "grid-gap"
+-- Sets both "gap" & "grid-gap"
+-- to quote: https://developer.mozilla.org/en-US/docs/Web/CSS/gap
+-- "CSS Grid Layout initially defined the grid-gap property. This prefixed property is being replaced by gap. However, in order to support browsers that implemented grid-gap and not gap for grid, you will need to use the prefixed property"
+gap :: Size a -> Css
+gap = key "gap" <> key "grid-gap"
+
+-- | Property sets the size of the gap (gutter) between an element's grid rows.
+-- Sets both "row-gap" & "grid-row-gap"
+-- to quote: https://developer.mozilla.org/en-US/docs/Web/CSS/row-gap
+-- "CSS Grid Layout initially defined the grid-row-gap property. This prefixed property is being replaced by row-gap. However, in order to support browsers that implemented grid-row-gap and not row-gap for grid, you will need to use the prefixed property."
+rowGap :: Size a -> Css
+rowGap = key "row-gap" <> key "grid-row-gap"
+
+-- | Property sets the size of the gap (gutter) between an element's grid columns.
+-- Sets both "column-gap" & "grid-column-gap"
+-- to quote: https://developer.mozilla.org/en-US/docs/Web/CSS/column-gap
+-- "CSS Grid Layout initially defined the grid-column-gap property. This prefixed property is being replaced by column-gap. However, in order to support bcolumnsers that implemented grid-column-gap and not column-gap for grid, you will need to use the prefixed property."
+columnGap :: Size a -> Css
+columnGap = key "column-gap" <> key "grid-column-gap"
+
+-------------------------------------------------------------------------------
+
+-- | Property defines the line names and track sizing functions of the grid rows.
+gridTemplateRows :: GridTrackList a -> Css
+gridTemplateRows = key "grid-template-rows"
+
+-- | Property defines the line names and track sizing functions of the grid columns.
+gridTemplateColumns :: GridTrackList a -> Css
+gridTemplateColumns = key "grid-template-columns"
+
+newtype GridTrackList a = GridTrackList Value
+  deriving (Val, None, Inherit, Initial, Unset)
+
+-- | Create a GridTrackList from a list of sizes
+mkGridTrackList :: [Size a] -> GridTrackList a
+mkGridTrackList = GridTrackList . noCommas
+
+-------------------------------------------------------------------------------
+
+-- | Property defines the line names and track sizing functions of the grid rows.
+gridAutoRows :: GridAutoTrackList a -> Css
+gridAutoRows = key "grid-auto-rows"
 
 -- | Property defines the line names and track sizing functions of the grid columns.
-gridTemplateColumns :: [Size a] -> Css
-gridTemplateColumns = key "grid-template-columns" . noCommas
+gridAutoColumns :: GridAutoTrackList a -> Css
+gridAutoColumns = key "grid-auto-columns"
+
+newtype GridAutoTrackList a = GridAutoTrackList Value
+  deriving (Val, Auto, MinContent, MaxContent, Inherit, Initial, Unset)
```

Comment:
```suggestion
  deriving (Val, Auto, MinContent, MaxContent)
```
I think the others are not valid, see https://www.w3schools.com/cssref/pr_grid-auto-columns.asp